Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp

```diff
@@ -1421,8 +1421,13 @@ MediaTime MediaPlayerPrivateGStreamer::playbackPosition() const
     GST_TRACE_OBJECT(pipeline(), "isEndReached: %s, seeking: %s, seekTime: %s", boolForPrinting(m_isEndReached), boolForPrinting(m_isSeeking), m_seekTime.toString().utf8().data());
     if (m_isSeeking)
         return m_seekTime;
-    if (m_isEndReached)
-        return m_playbackRate > 0 ? durationMediaTime() : MediaTime::zeroTime();
+    if (m_isEndReached) {
+        if (m_playbackRate > 0) {
+            MediaTime duration = durationMediaTime();
+            return (m_cachedPosition.isValid() && m_cachedPosition > duration) ? m_cachedPosition : duration;
```

Comment:
I think this goes against [the spec](https://html.spec.whatwg.org/multipage/media.html#offsets-into-the-media-resource), which says:

> If an "infinite" stream ends for some reason, then the duration would change from positive Infinity to the time of the last frame or sample in the stream, and the [durationchange](https://html.spec.whatwg.org/multipage/media.html#event-media-durationchange) event would be fired. Similarly, if the user agent initially estimated the [media resource](https://html.spec.whatwg.org/multipage/media.html#media-resource)'s duration instead of determining it precisely, and later revises the estimate based on new information, then the duration would change and the [durationchange](https://html.spec.whatwg.org/multipage/media.html#event-media-durationchange) event would be fired.

In which case would the cached position be higher than the duration? Have into account that [the duration has already been updated to the latest position when didEnd() happened](https://github.com/WebPlatformForEmbedded/WPEWebKit/blob/wpe-2.28/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp#L2717).

Reply:
Indeed, it works fine with the duration update done in didEnd(). I've removed this change.